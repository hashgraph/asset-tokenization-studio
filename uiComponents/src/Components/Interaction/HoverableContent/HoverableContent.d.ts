import type { FlexProps } from "@chakra-ui/react";
import type { ReactNode } from "react";
export interface HoverableContentProps extends FlexProps {
    hiddenContent: ReactNode;
}
export declare const HoverableContent: ({ children, hiddenContent, ...props }: HoverableContentProps) => JSX.Element;
