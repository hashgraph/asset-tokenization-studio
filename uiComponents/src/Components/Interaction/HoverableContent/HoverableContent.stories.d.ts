declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./HoverableContent").HoverableContentProps>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./HoverableContent").HoverableContentProps>;
