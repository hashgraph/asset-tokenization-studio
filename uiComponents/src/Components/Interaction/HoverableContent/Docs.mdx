import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as HoverableContentStories from "./HoverableContent.stories";

<Meta of={HoverableContentStories} />

<Title />

<Primary />

## Props

<ArgTypes of={HoverableContentStories} />

## Stories

<Stories title="" />
