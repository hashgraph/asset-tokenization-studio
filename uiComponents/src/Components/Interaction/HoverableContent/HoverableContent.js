var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { useHover } from "@/Hooks";
import { Flex } from "@chakra-ui/react";
import React from "react";
export var HoverableContent = function (_a) {
    var children = _a.children, hiddenContent = _a.hiddenContent, props = __rest(_a, ["children", "hiddenContent"]);
    var _b = useHover(), containerRef = _b[0], isHovered = _b[1], refHover = _b[2];
    var handleEvent = function (eventName) {
        if (refHover.current) {
            refHover.current.dispatchEvent(new Event(eventName));
        }
    };
    return (React.createElement(Flex, __assign({ ref: containerRef, "data-testid": "hoverable-content", tabIndex: 0, onFocus: function () { return handleEvent("mouseenter"); }, onBlur: function () { return handleEvent("mouseleave"); }, _focusVisible: { outline: "var(--chakra-colors-primary-100) auto 1px" } }, props),
        children,
        isHovered && hiddenContent));
};
