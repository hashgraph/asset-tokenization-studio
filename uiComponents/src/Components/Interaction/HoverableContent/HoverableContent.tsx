import { useHover } from "@/Hooks";
import type { FlexProps } from "@chakra-ui/react";
import { Flex } from "@chakra-ui/react";
import type { ReactNode } from "react";
import React from "react";

export interface HoverableContentProps extends FlexProps {
  hiddenContent: ReactNode;
}

export const HoverableContent = ({
  children,
  hiddenContent,
  ...props
}: HoverableContentProps) => {
  const [containerRef, isHovered, refHover] = useHover();

  const handleEvent = (eventName: string) => {
    if (refHover.current) {
      refHover.current.dispatchEvent(new Event(eventName));
    }
  };

  return (
    <Flex
      ref={containerRef}
      data-testid="hoverable-content"
      tabIndex={0}
      onFocus={() => handleEvent("mouseenter")}
      onBlur={() => handleEvent("mouseleave")}
      _focusVisible={{ outline: "var(--chakra-colors-primary-100) auto 1px" }}
      {...props}
    >
      {children}
      {isHovered && hiddenContent}
    </Flex>
  );
};
