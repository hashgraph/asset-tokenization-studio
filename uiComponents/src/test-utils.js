import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
