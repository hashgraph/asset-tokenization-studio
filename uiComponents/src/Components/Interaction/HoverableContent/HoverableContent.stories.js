var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { HoverableContent } from "./HoverableContent";
import React from "react";
import { Box } from "@chakra-ui/react";
var meta = {
    title: "Design System/Interaction/HoverableContent",
    component: HoverableContent,
    args: {
        hiddenContent: React.createElement(Box, { ml: 6 }, "Hello!")
    },
    parameters: {}
};
export default meta;
var Template = function (args) { return (React.createElement(HoverableContent, __assign({}, args),
    React.createElement(Box, null, "Hover to display content"))); };
export var Default = Template.bind({});
Default.args = {};
