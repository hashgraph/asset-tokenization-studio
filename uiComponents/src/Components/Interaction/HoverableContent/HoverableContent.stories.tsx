import type { Meta, StoryFn } from "@storybook/react";
import { HoverableContent } from "./HoverableContent";
import React from "react";
import { Box } from "@chakra-ui/react";

const meta = {
  title: "Design System/Interaction/HoverableContent",
  component: HoverableContent,
  args: {
    hiddenContent: <Box ml={6}>Hello!</Box>,
  },
  parameters: {},
} as Meta<typeof HoverableContent>;
export default meta;

const Template: StoryFn<typeof HoverableContent> = (args) => (
  <HoverableContent {...args}>
    <Box>Hover to display content</Box>
  </HoverableContent>
);

export const Default = Template.bind({});
Default.args = {};
