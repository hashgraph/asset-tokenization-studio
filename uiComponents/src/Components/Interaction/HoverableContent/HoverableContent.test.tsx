import { render } from "@/test-utils";
import React from "react";
import type { HoverableContentProps } from "./HoverableContent";
import { HoverableContent } from "./HoverableContent";
import { Box } from "@chakra-ui/react";
import { fireEvent } from "@testing-library/react";

describe(`<HoverableContent />`, () => {
  const defaultProps = {
    hiddenContent: "Hover",
  };

  const factoryComponent = (props: HoverableContentProps = defaultProps) =>
    render(
      <HoverableContent {...props}>
        <Box>Children</Box>
      </HoverableContent>
    );

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders content when children is hovered", () => {
    const component = factoryComponent();
    const wrapper = component.getByTestId("hoverable-content");

    expect(component.queryByText("Hover")).not.toBeInTheDocument();

    fireEvent.mouseEnter(wrapper);

    expect(component.queryByText("Hover")).toBeInTheDocument();
  });
});
