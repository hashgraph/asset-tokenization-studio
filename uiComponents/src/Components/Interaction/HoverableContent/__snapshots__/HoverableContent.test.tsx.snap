// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HoverableContent /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1en4ag4"
    data-testid="hoverable-content"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      Children
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
